src/DiIiS-NA/Core/Discord/Bot.cs

```diff
@@ -225,10 +226,9 @@ public async Task FinishGiveaway()
 						var message = await guild.GetTextChannel(channelId).GetMessageAsync(param.First().Value);
 						var reactedUsers = await (message as IUserMessage).GetReactionUsersAsync(Emote.Parse("<:wolfRNG:607868292979490816>"), 100).FlattenAsync();
 						var contestants = reactedUsers.Where(u => !u.IsBot).ToArray();
-						
-						if (contestants.Length > 0)
+
+						if (contestants.TryPickRandom(out var winner))
 						{
-							var winner = contestants[Helpers.Math.FastRandom.Instance.Next(0, contestants.Length - 1)];
```

Comment:
btw, this FastRandom is not so fast. It is slower 20-30% than built-in one and I am not sure it is thread safe